server/graphql/v2/mutation/OrderMutations.js

```diff
@@ -315,6 +316,31 @@ const orderMutations = {
           throw new ValidationFailed(`Only pending/expired orders can be marked as paid, this one is ${order.status}`);
         }
 
+        const hasUpdates = !isEmpty(difference(keys(args.order), ['id', 'legacyId']));
+        if (hasUpdates) {
+          const { amount, paymentProcessorFee, platformTip } = args.order;
+
+          if (amount) {
+            const totalAmount = amount + platformTip;
+            order.set('totalAmount', totalAmount);
+          }
+          if (paymentProcessorFee) {
+            if (!order.data) {
+              order.set('data', {});
+            }
+            order.set('data.paymentProcessorFee', paymentProcessorFee);
+          }
+          if (platformTip) {
+            if (!order.data) {
+              order.set('data', {});
+            }
+            order.set('data.platformTip', platformTip);
+            // Some parts of the order flow still uses data.platformFee
+            order.set('data.platformFee', platformTip);
```

Comment:
Cool, setting the two is a good way to anticipate a future migration.

Reply:
Yes, I think it is time for us to create a proper column to track the platform contribution and clean this tech debt.